app/src/main/java/de/tum/informatics/www1/artemis/native_app/android/ui/MainActivity.kt

```diff
@@ -347,7 +350,7 @@ class MainActivity : AppCompatActivity(),
                 quizParticipation(
                     onLeaveQuiz = {
                         val previousBackStackEntry = navController.previousBackStackEntry
-                        if (previousBackStackEntry?.destination?.route == ExerciseViewDestination.EXERCISE_VIEW_ROUTE) {
+                        if (previousBackStackEntry?.destination?.route == ExerciseViewUi::class.qualifiedName.orEmpty()) {
```

Comment:
Did you verify test this code?

Reply:
We have currently disabled Quiz participation, see https://github.com/ls1intum/artemis-android/issues/107. I will add this check as a todo there